modules/imgcodecs/test/test_exr.impl.hpp

```diff
@@ -113,5 +113,151 @@ TEST(Imgcodecs_EXR, readWrite_32FC3_half)
     EXPECT_EQ(0, remove(filenameOutput.c_str()));
 }
 
+// Note: YC to GRAYSCALE (IMREAD_GRAYSCALE | IMREAD_ANYDEPTH)
+// outputs a black image,
+// as does Y to RGB (IMREAD_COLOR | IMREAD_ANYDEPTH).
+// This behavoir predates adding EXR alpha support issue
+// 16115.
+
+TEST(Imgcodecs_EXR, read_YA_ignore_alpha)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filenameInput = root + "readwrite/test_YA.exr";
+
+    const Mat img = cv::imread(filenameInput, IMREAD_GRAYSCALE | IMREAD_ANYDEPTH);
+
+    ASSERT_FALSE(img.empty());
+    ASSERT_EQ(CV_32FC1, img.type());
+
+    // Writing Y covered by test 32FC1
+}
+
+TEST(Imgcodecs_EXR, read_YA_unchanged)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filenameInput = root + "readwrite/test_YA.exr";
+
+    const Mat img = cv::imread(filenameInput, IMREAD_UNCHANGED);
+
+    ASSERT_FALSE(img.empty());
+    ASSERT_EQ(CV_32FC2, img.type());
+
+    // Cannot test writing, 2 channel writing not suppported by loadsave
+}
+
+TEST(Imgcodecs_EXR, read_YC_changeDepth)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filenameInput = root + "readwrite/test_YRYBY.exr";
+
+    const Mat img = cv::imread(filenameInput, IMREAD_COLOR);
+
+    ASSERT_FALSE(img.empty());
+    ASSERT_EQ(CV_8UC3, img.type());
+
+    // Cannot test writing, EXR encoder doesn't support 8U depth
+}
+
+TEST(Imgcodecs_EXR, readwrite_YCA_ignore_alpha)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filenameInput = root + "readwrite/test_YRYBYA.exr";
+    const string filenameOutput = root + "readwrite/test_YRYBYA.written.removedalpha.exr";
```

Comment:
Tests should not write to `testdata` location by default. It may be mounted in read-only mode (and shared between validation build instances).

Consider using `GENERATE_DATA` approach to generate test data (write to testdata location "locally" and then commit generated files).
Use `cv::tempfile()` to generate names for temporary files.

You need to open PR in [opencv_extra](https://github.com/opencv/opencv_extra) (with the same source branch name `EXR_rw_alpha_support_16115`), not in your fork.

Reply:
Thank you @alalek.  Good points.  I see the issue with writing to testdata location.  I updated to write to cv::tempfile() location & name.

I updated the code to use GENERATE_DATA for the RGBA test case (and committed a generated RGBA test image to opencv_extra).  GENERATE_DATA will not work for YA or YRYBY(A) test images as the current EXR write only supports RGB(A) or Y (all input chroma files are convereted to RGB(A) or Y(A) on read), and loadsave doesn't support YA (2 channel) write.

I believe I updated the opencv_extra PR as instructed: https://github.com/opencv/opencv_extra/pull/859

Reply:
Hi @alalek.  Are there further changes to my PR you would like to see?
Thank you.